### PR TITLE
fix(material/snack-bar): incorrect width at some breakpoints

### LINKED ISSUE #26290: bug(snack-bar): MDC media query not correctly working
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

14

### Description

Since v15, MDC snack bar is used and it provides a breakpoint at 480px which overrides `mdc-snackbar__surface`'s `min-width` to `100%` but it doesn't work correctly for Material Angular snack bars where it seems to take min content width instead of full viewport width.

Also, just a thought: is `handsetCssClass` still relevant now there is already this MDC breakpoint? (it's seems redundant)

### Reproduction

All the behaviors described below are achieved using Chrome's device toolbar with responsive dimensions of 480x400.

### Expected Behavior

When looking at the example from [MDC Web](https://material-components.github.io/material-components-web-catalog/#/component/snackbar), the `min-width` is working as expected (by taking the full width of the viewport):
material-components github io_material-components-web-catalog_ (3)

Just for reference, here's how it was working in v14 (correctly because no MDC breakpoint):
wrong-snackbar-action-color-with-mat-app-background-14 stackblitz io_ (2)
(https://stackblitz.com/edit/wrong-snackbar-action-color-with-mat-app-background-14?file=src%2Fapp%2Fapp.component.ts)

### Actual Behavior

Instead of taking the full width of the viewport like in the MDC Web example, in v15, the Angular Material snack bar seems to take min content width:
wrong-snackbar-action-color-with-mat-app-backgrou-39aglb stackblitz io_
(https://stackblitz.com/edit/wrong-snackbar-action-color-with-mat-app-background-15?file=src%2Fapp%2Fapp.component.ts)

### Environment

- Angular: 15.0.3
- CDK/Material: 15.0.2
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

### PR DESCRIPTION
**Pull request by crisbeto**
MDC has some built in breakpoints where the snack bar container becomes `min-width: 100%`. They didn't work correctly, because the container wasn't set up to be 100% wide.

Fixes #26290.

### CODE REVIEW

File: src/material/snack-bar/snack-bar-container.scss [hunk 1]
```diff
@@ -60,10 +74,6 @@
   // of positions, so we'll defer logic there.
   position: static;
 
-  @include cdk.high-contrast(active, off) {
```
Comment (crisbeto) [inline src/material/snack-bar/snack-bar-container.scss:63]: Unrelated change. I just noticed that we don't need it anymore.

File: src/material/snack-bar/snack-bar-container.scss [hunk 2]
```diff
@@ -10,13 +9,28 @@
...
+  $mobile-breakpoint: mdc-snackbar-theme.$mobile-breakpoint;
+
+  // The styles weren't included in `static-styles` so we need to add them ourselves.
+  @include mdc-snackbar-theme.min-width(
+    $min-width: $min-width,
+    $mobile-breakpoint: $mobile-breakpoint,
+    $query: mdc-helpers.$mdc-base-styles-query
+  );
+
+  // The MDC `min-width` mixin has a similar breakpoint that sets `min-width: 100%` on the surfcace
```
Comment (jelbourn) [inline src/material/snack-bar/snack-bar-container.scss:23]:
```suggestion
  // The MDC `min-width` mixin has a similar breakpoint that sets `min-width: 100%` on the surface
```

File: src/material/snack-bar/snack-bar-container.scss [hunk 3]
```diff
@@ -10,13 +9,28 @@
...
+  // element to make it span the entire viewport, however it ends up collapsing because the
+  // container is `width: auto`. This query ensures that the surface will span the whole viewport.
```
Comment (jelbourn) [inline src/material/snack-bar/snack-bar-container.scss:25]: Should we open an issue OR code change with MDC for this? It seems like `100vw` more closely matches the intent of their style

Comment (crisbeto) [reply src/material/snack-bar/snack-bar-container.scss:25]: MDC would've been 100vw as well, but we override their `position: fixed` positioning in order to fit the snack bar in our overlay positioning model.
